langextract/inference.py

```diff
@@ -298,10 +307,13 @@ def __init__(
     self.max_workers = max_workers
     self._extra_kwargs = kwargs or {}
 
-    if not self.api_key:
-      raise ValueError('API key not provided.')
-
-    self._client = genai.Client(api_key=self.api_key)
+    self._client = genai.Client(
+        api_key=self.api_key,
+        vertexai=vertexai,
+        credentials=credentials,
+        project=project,
+        location=location,
+    )
```

Comment:
I'm super interested in this ability too, given for enterprise SA's are how our Platform Engineering team has allowed access to VertexAI (over api_key). 

However, I do have one ask if you can add - we use VPC endpoints, so adding the http_options block would also be needed to allow overriding the endpoint used - see below code;

`import google.genai as genai

client = genai.Client(
    project="<project_id>",
    location="<region>",
    vertexai=True,
    http_options={
        "base_url": "https://<vpc_endpoint>.p.googleapis.com"
    }
)`

Reply:
@JamesWordie That would indeed make sense!

Since I already silenced a lint check with this, adding yet another attribute seems weird, so I'm thinking of adding a `VertexAIConfig` dataclass with the `credentials`, `project`, `location`, and now `http_options` fields to replace all the currently added fields with a `vertex_ai_config: VertexAIConfig` field.

Thoughts?